### ISSUE TITLE
Unable use altherma in automation

### ISSUE THREAD

**Issue by palecs** [speleolontra/daikin_residential_altherma#82]
Hello I dont know if Im making something wrong, but I went through the topics, through internet, but I would like to make automation like this - if there is exces of solar energy set tank temperature higher and heat water in tank. I can make everything with solar, but im not able to set the temperature in altherma. I have tried to disable tank plan and weather heating, but climate.set_temperature is not working. error is regarding not mapped to room/zone/etc. 

For next features I would also like to set heating offset in room heating etc. I also dont see to ability (gauge) to set temperature. 

Thank you for your help, superb work, great integration.

**Comment by palecs** [comment]
Im also attaching few logs. When I reload this add-on Im getting this error: 

Logger: homeassistant.components.climate
Source: custom_components/daikin_residential_altherma/daikin_base.py:300
Integration: Climate ([documentation](https://www.home-assistant.io/integrations/climate), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+climate%22))
First occurred: 18:15:17 (8 occurrences)
Last logged: 18:52:48

Error adding entities for domain climate with platform daikin_residential_altherma
Error while setting up daikin_residential_altherma platform for climate
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 442, in async_add_entities
    await asyncio.gather(*tasks)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 629, in _async_add_entity
    capabilities=entity.capability_attributes,
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 264, in capability_attributes
    self.hass, self.min_temp, self.temperature_unit, self.precision
  File "/config/custom_components/daikin_residential_altherma/climate.py", line 191, in min_temp
    return self._device.min_temp
  File "/config/custom_components/daikin_residential_altherma/daikin_base.py", line 300, in min_temp
    return float(self.getData(ATTR_TARGET_ROOM_TEMPERATURE)["minValue"])
TypeError: 'NoneType' object is not subscriptable


Diagnostic info are like this:
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2023.2.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.10.7",
    "docker": true,
    "arch": "armv7l",
    "timezone": "Europe/Bratislava",
    "os_name": "Linux",
    "os_version": "5.15.84-v7l",
    "supervisor": "2023.01.1",
    "host_os": "Home Assistant OS 9.5",
    "docker_version": "20.10.22",
    "chassis": "embedded",
    "run_as_root": true
  },
  "custom_components": {
    "hacs": {
      "version": "1.30.1",
      "requirements": [
        "aiogithubapi>=22.10.1"
      ]
    },
    "daikin_residential_altherma": {
      "version": "2.1.0",
      "requirements": [
        "oic==1.4.0"
      ]
    },
    "eldes_alarm": {
      "version": "1.0.0",
      "requirements": []
    },
    "huawei_solar": {
      "version": "1.2.4",
      "requirements": [
        "huawei-solar==2.2.4"
      ]
    }
  },
  "integration_manifest": {
    "domain": "daikin_residential_altherma",
    "name": "Daikin Residential Controller for Altherma",
    "version": "2.1.0",
    "documentation": "https://github.com/speleolontra/daikin_residential_altherma",
    "dependencies": [],
    "codeowners": [
      "@speleolontra"
    ],
    "issue_tracker": "https://github.com/speleolontra/daikin_residential_altherma/issues",
    "requirements": [
      "oic==1.4.0"
    ],
    "iot_class": "cloud_polling",
    "config_flow": true,
    "is_built_in": false
  },
  "data": {
    "tokenset": {
      "email": "XXXXXXX",
      "password": "XXXXXXXX",
      "tokenset": {
        "access_token": "XXX",
        "refresh_token": "XXXX",
        "expires_in": 3600,
        "token_type": "Bearer"
      }
    },
    "json_data": [
      {
        "_id": "88987bd9-a7aa-4ca7-a65f-754ffd0fda08",
        "deviceModel": "Altherma",
        "type": "heating-wlan",
        "isCloudConnectionUp": {
          "settable": false,
          "value": true
        },
        "managementPoints": [
          {
            "embeddedId": "gateway",
            "managementPointType": "gateway",
            "managementPointCategory": "secondary",
            "firmwareVersion": {
              "settable": false,
              "value": "3.2.4",
              "maxLength": 8
            },
            "ipAddress": {
              "settable": false,
              "value": "192.168.4.201",
              "maxLength": 15
            },
            "iconId": {
              "settable": true,
              "requiresReboot": false,
              "value": 3
            },
            "isFirmwareUpdateSupported": {
              "settable": false,
              "requiresReboot": false,
              "value": true
            },
            "macAddress": {
              "settable": false,
              "value": "00:e9:3a:4b:16:50",
              "maxLength": 17
            },
            "modelInfo": {
              "settable": false,
              "value": "BRP069A78",
              "maxLength": 9
            },
            "name": {
              "settable": true,
              "requiresReboot": false,
              "value": "Gateway",
              "maxLength": 63
            },
            "ssid": {
              "settable": false,
              "requiresReboot": false,
              "value": "daikin-ap",
              "maxLength": 9
            },
            "serialNumber": {
              "settable": false,
              "value": "0019465",
              "maxLength": 16
            },
            "wifiConnectionSSID": {
              "settable": false,
              "requiresReboot": false,
              "value": "zIoT",
              "maxLength": 32
            },
            "wifiConnectionStrength": {
              "settable": false,
              "requiresReboot": false,
              "value": -36,
              "maxValue": 0,
              "minValue": -90,
              "stepValue": 1
            }
          },
          {
            "embeddedId": "climateControlMainZone",
            "managementPointType": "climateControl",
            "managementPointCategory": "primary",
            "managementPointSubType": "mainZone",
            "consumptionData": {
              "settable": false,
              "requiresReboot": false,
              "ref": "#consumptionData",
              "value": {
                "electrical": {
                  "heating": {
                    "d": [
                      2,
                      2,
                      1,
                      2,
                      1,
                      2,
                      1,
                      1,
                      2,
                      2,
                      1,
                      1,
                      1,
                      1,
                      2,
                      1,
                      2,
                      1,
                      2,
                      1,
                      1,
                      null,
                      null,
                      null
                    ],
                    "w": [
                      30,
                      20,
                      11,
                      12,
                      13,
                      17,
                      20,
                      19,
                      20,
                      18,
                      12,
                      null,
                      null,
                      null
                    ],
                    "m": [
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      817,
                      142,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null
                    ]
                  },
                  "cooling": {
                    "d": [
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      null,
                      null,
                      null
                    ],
                    "w": [
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      0,
                      null,
                      null,
                      null
                    ],
                    "m": [
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      0,
                      0,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null
                    ]
                  }
                }
              }
            },
            "controlMode": {
              "settable": false,
              "requiresReboot": false,
              "value": "leavingWaterTemperature",
              "values": [
                "leavingWaterTemperature",
                "externalRoomTemperature",
                "roomTemperature"
              ]
            },
            "errorCode": {
              "settable": false,
              "requiresReboot": false,
              "value": "-00",
              "maxLength": 16
            },
            "holidayMode": {
              "settable": true,
              "requiresReboot": false,
              "ref": "#holidayMode",
              "value": {
                "enabled": false,
                "startDate": "2022-04-16",
                "endDate": "2022-04-20"
              }
            },
            "iconId": {
              "settable": true,
              "requiresReboot": false,
              "value": 8
            },
            "isHolidayModeActive": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "isInEmergencyState": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "isInErrorState": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "isInInstallerState": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "isInWarningState": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "name": {
              "settable": true,
              "requiresReboot": false,
              "value": "",
              "maxLength": 63
            },
            "onOffMode": {
              "settable": true,
              "requiresReboot": false,
              "value": "on",
              "values": [
                "off",
                "on"
              ]
            },
            "operationMode": {
              "settable": true,
              "requiresReboot": false,
              "value": "heating",
              "values": [
                "heating",
                "cooling",
                "auto"
              ]
            },
            "sensoryData": {
              "settable": false,
              "ref": "#sensoryData",
              "value": {
                "roomTemperature": {
                  "settable": false,
                  "requiresReboot": false,
                  "value": 23,
                  "maxValue": 127,
                  "minValue": -127,
                  "stepValue": 0.1
                },
                "outdoorTemperature": {
                  "settable": false,
                  "requiresReboot": false,
                  "value": 0,
                  "maxValue": 127,
                  "minValue": -127,
                  "stepValue": 1
                },
                "leavingWaterTemperature": {
                  "settable": false,
                  "requiresReboot": false,
                  "value": 29,
                  "maxValue": 127,
                  "minValue": -127,
                  "stepValue": 1
                }
              }
            },
            "setpointMode": {
              "settable": false,
              "requiresReboot": true,
              "value": "weatherDependent",
              "values": [
                "fixed",
                "weatherDependentHeatingFixedCooling",
                "weatherDependent"
              ]
            },
            "temperatureControl": {
              "settable": true,
              "ref": "#temperatureControl",
              "value": {
                "operationModes": {
                  "auto": {
                    "setpoints": {
                      "leavingWaterOffset": {
                        "settable": true,
                        "requiresReboot": false,
                        "value": -5,
                        "maxValue": 10,
                        "minValue": -10,
                        "stepValue": 1
                      }
                    }
                  },
                  "heating": {
                    "setpoints": {
                      "leavingWaterOffset": {
                        "settable": true,
                        "requiresReboot": false,
                        "value": -5,
                        "maxValue": 10,
                        "minValue": -10,
                        "stepValue": 1
                      }
                    }
                  },
                  "cooling": {
                    "setpoints": {
                      "leavingWaterOffset": {
                        "settable": true,
                        "requiresReboot": false,
                        "value": 0,
                        "maxValue": 10,
                        "minValue": -10,
                        "stepValue": 1
                      }
                    }
                  }
                }
              }
            }
          },
          {
            "embeddedId": "domesticHotWaterTank",
            "managementPointType": "domesticHotWaterTank",
            "managementPointCategory": "primary",
            "consumptionData": {
              "settable": false,
              "requiresReboot": false,
              "ref": "#consumptionData",
              "value": {
                "electrical": {
                  "heating": {
                    "d": [
                      0,
                      0,
                      0,
                      3,
                      0,
                      0,
                      0,
                      0,
                      0,
                      1,
                      0,
                      0,
                      0,
                      0,
                      0,
                      3,
                      0,
                      0,
                      0,
                      0,
                      0,
                      null,
                      null,
                      null
                    ],
                    "w": [
                      2,
                      3,
                      4,
                      4,
                      3,
                      4,
                      4,
                      4,
                      3,
                      4,
                      3,
                      null,
                      null,
                      null
                    ],
                    "m": [
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      84,
                      33,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null,
                      null
                    ]
                  }
                }
              }
            },
            "errorCode": {
              "settable": false,
              "requiresReboot": false,
              "value": "-00",
              "maxLength": 16
            },
            "iconId": {
              "settable": true,
              "requiresReboot": false,
              "value": 9
            },
            "isHolidayModeActive": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "isInEmergencyState": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "isInErrorState": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "isInInstallerState": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "isInWarningState": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "isPowerfulModeActive": {
              "settable": false,
              "requiresReboot": false,
              "value": false
            },
            "name": {
              "settable": true,
              "requiresReboot": false,
              "value": "",
              "maxLength": 63
            },
            "onOffMode": {
              "settable": true,
              "requiresReboot": false,
              "value": "on",
              "values": [
                "off",
                "on"
              ]
            },
            "operationMode": {
              "settable": false,
              "value": "heating",
              "values": [
                "heating"
              ]
            },
            "powerfulMode": {
              "settable": true,
              "requiresReboot": false,
              "value": "off",
              "values": [
                "off",
                "on"
              ]
            },
            "schedule": {
              "settable": true,
              "ref": "#schedule",
              "value": {
                "currentMode": {
                  "settable": false,
                  "value": "heating",
                  "values": [
                    "heating"
                  ]
                },
                "modes": {
                  "heating": {
                    "enabled": {
                      "settable": false,
                      "requiresReboot": false,
                      "value": true
                    },
                    "currentSchedule": {
                      "settable": true,
                      "requiresReboot": false,
                      "value": "scheduleHeatingMode1",
                      "values": [
                        "scheduleHeatingMode1"
                      ]
                    },
                    "meta": {
                      "minIntervalBetweenActions": "00:10:00",
                      "maxSchedules": 1,
                      "maxActionsPerActionPeriod": 4,
                      "consecutiveActionsAllowed": true,
                      "actionTypes": {
                        "domesticHotWaterTemperature": {
                          "settable": false,
                          "values": [
                            "eco",
                            "comfort",
                            "turn_off"
                          ]
                        }
                      }
                    },
                    "schedules": {
                      "scheduleHeatingMode1": {
                        "settable": true,
                        "name": {
                          "settable": true,
                          "requiresReboot": false,
                          "value": "User defined"
                        },
                        "meta": {
                          "isReadOnly": false,
                          "actionPeriods": [
                            "monday",
                            "tuesday",
                            "wednesday",
                            "thursday",
                            "friday",
                            "saturday",
                            "sunday"
                          ]
                        },
                        "actions": {
                          "monday": {
                            "05:50:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "07:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            },
                            "18:00:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "21:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            }
                          },
                          "tuesday": {
                            "05:50:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "07:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            },
                            "18:00:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "21:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            }
                          },
                          "wednesday": {
                            "05:50:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "07:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            },
                            "18:00:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "21:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            }
                          },
                          "thursday": {
                            "05:50:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "07:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            },
                            "18:00:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "21:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            }
                          },
                          "friday": {
                            "05:50:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "07:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            },
                            "18:00:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "21:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            }
                          },
                          "saturday": {
                            "05:50:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "07:30:00": {
                              "domesticHotWaterTemperature": "eco"
                            },
                            "18:00:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "21:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            }
                          },
                          "sunday": {
                            "05:50:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "07:30:00": {
                              "domesticHotWaterTemperature": "eco"
                            },
                            "17:30:00": {
                              "domesticHotWaterTemperature": "comfort"
                            },
                            "21:00:00": {
                              "domesticHotWaterTemperature": "turn_off"
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            },
            "sensoryData": {
              "settable": false,
              "ref": "#sensoryData",
              "value": {
                "tankTemperature": {
                  "settable": false,
                  "requiresReboot": false,
                  "value": 43,
                  "maxValue": 127,
                  "minValue": -127,
                  "stepValue": 1
                }
              }
            },
            "setpointMode": {
              "settable": false,
              "requiresReboot": false,
              "value": "fixed",
              "values": [
                "fixed",
                "weatherDependent"
              ]
            },
            "temperatureControl": {
              "settable": true,
              "ref": "#temperatureControl",
              "value": {
                "operationModes": {
                  "heating": {
                    "setpoints": {
                      "domesticHotWaterTemperature": {
                        "settable": false,
                        "requiresReboot": false,
                        "value": 43,
                        "maxValue": 50,
                        "minValue": 30,
                        "stepValue": 1
                      }
                    }
                  }
                }
              }
            }
          },
          {
            "embeddedId": "indoorUnitHydro",
            "managementPointType": "indoorUnitHydro",
            "managementPointCategory": "secondary",
            "eepromVersion": {
              "settable": false,
              "requiresReboot": false,
              "value": "3608726-74",
              "maxLength": 16
            },
            "iconId": {
              "settable": true,
              "requiresReboot": false,
              "value": 4
            },
            "modelInfo": {
              "settable": false,
              "requiresReboot": false,
              "value": "EHVX08S23EJ9W",
              "maxLength": 16
            },
            "name": {
              "settable": true,
              "requiresReboot": false,
              "value": "Indoor Hydro Unit",
              "maxLength": 63
            },
            "softwareVersion": {
              "settable": false,
              "requiresReboot": false,
              "value": "0221",
              "maxLength": 16
            }
          },
          {
            "embeddedId": "outdoorUnit",
            "managementPointType": "outdoorUnit",
            "managementPointCategory": "secondary",
            "iconId": {
              "settable": true,
              "requiresReboot": false,
              "value": 5
            },
            "modelInfo": {
              "settable": false,
              "requiresReboot": false,
              "value": "",
              "maxLength": 16
            },
            "name": {
              "settable": true,
              "requiresReboot": false,
              "value": "Outdoor Unit",
              "maxLength": 63
            },
            "softwareVersion": {
              "settable": false,
              "requiresReboot": false,
              "value": "FFFF",
              "maxLength": 16
            }
          },
          {
            "embeddedId": "userInterface",
            "managementPointType": "userInterface",
            "managementPointCategory": "secondary",
            "dateTime": {
              "settable": false,
              "requiresReboot": false,
              "value": "2023-02-09T10:26:43"
            },
            "firmwareVersion": {
              "settable": false,
              "requiresReboot": false,
              "deprecated": "DEPRECATED",
              "value": "6.3.0",
              "maxLength": 16
            },
            "iconId": {
              "settable": true,
              "requiresReboot": false,
              "value": 6
            },
            "miconId": {
              "settable": false,
              "requiresReboot": false,
              "value": "20010E02",
              "maxLength": 16
            },
            "name": {
              "settable": true,
              "requiresReboot": false,
              "value": "User Interface",
              "maxLength": 63
            },
            "softwareVersion": {
              "settable": false,
              "requiresReboot": false,
              "value": "6.3.0",
              "maxLength": 16
            }
          }
        ],
        "embeddedId": "6bfe37aa-43e0-4c23-93de-67cb2e34834a",
        "timestamp": "2023-02-09T17:59:03.566Z",
        "id": "88987bd9-a7aa-4ca7-a65f-754ffd0fda08",
        "lastUpdateReceived": "2023-02-09T17:59:03.566Z"
      }
    ]
  }
}

**Comment by jwillemsen** [comment]
It looks you can't set the `roomTemperature` to daikin, but you probably can set `leavingWaterOffset`. Does it work when you try to change the value within the HA UI?

**Comment by palecs** [comment]
Hello, after some playing with this code I maybe found the issue. My installation is heating based on water heating offset and I have also home termostat/temperature sensor, but not using it, only for displaying actual temperature of important room. If I play with logic in daikin_base.py - especially setting minimal and maximal temperature (1st check leavingWaterOffset, then roomTemperature), it starts correctly and also displays in widget the ability to change temp and swich mode etc. But changing temp in widget fails with error below, because it is setting temperature not water offset I guess. 

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 200, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1787, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1824, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 680, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 968, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 720, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 613, in async_service_temperature_set
    await entity.async_set_temperature(**kwargs)
  File "/config/custom_components/daikin_residential_altherma/climate.py", line 218, in async_set_temperature
    await self._device.async_set_temperature(kwargs[ATTR_TEMPERATURE])
  File "/config/custom_components/daikin_residential_altherma/daikin_base.py", line 354, in async_set_temperature
    return await self.setValue(ATTR_TARGET_ROOM_TEMPERATURE, value)
  File "/config/custom_components/daikin_residential_altherma/daikin_base.py", line 168, in setValue
    return await self.set_data(cmd_set[0], cmd_set[1], cmd_set[2], value)
  File "/config/custom_components/daikin_residential_altherma/device.py", line 391, in set_data
    raise Exception(
Exception: Please provide a valid datapoint definition that exists in the data structure

**Comment by palecs** [comment]
> It looks you can't set the `roomTemperature` to daikin, but you probably can set `leavingWaterOffset`. Does it work when you try to change the value within the HA UI?

Hello unfortunatelly im not able to set `leavingWaterOffset` in climate. As I mentioned in post below, I think issue is that logic of script is not considering `temperature sensor` and `leavingwateroffset` in one installation. What do you think can you fix/extend fuctionality? 👯

**Comment by palecs** [comment]
Hello I was able to fix this. All changes are made only in daikin_base.py. setting everything around temperature should be 1st `leavingWaterOffset` then `roomTemperature` but it is only temp fix. I guess there should be more control if there is thermostat, wateroffset mode and based on this set all parameters. If needed I can paste code. But I think it should have proper implementation. Because if I switch off waterOffset and start using temperature, I will face same issues, like now I guess, because I see some logic issues in logic in this setup.

Thank you.

**Comment by jwillemsen** [comment]
Hi @palecs, what is the status of this issue?

**Comment by jwillemsen** [comment]
github master now has support to set the water tank temperature

**Comment by jwillemsen** [comment]
Can we close this one now @palecs ?

**Comment by palecs** [comment]
Hello for now we can close this one. I will try it today and if there is any issue I will reopen it, or add new one with new issues.

**Comment by palecs** [comment]
By testing this im getting this errors in trying to set temperature in automation.

```
Logger: homeassistant.components.automation.export_teplota_bojler
Source: components/automation/__init__.py:255
Integration: Automation (documentation, issues)
First occurred: 18:04:13 (6 occurrences)
Last logged: 18:41:36

Error while executing automation automation.export_teplota_bojler: extra keys not allowed @ data['area_id']
Error while executing automation automation.export_teplota_bojler: extra keys not allowed @ data['device_id']
```

Details of automation is:

```
alias: Export teplota bojler
description: ""
trigger:
  - platform: numeric_state
    entity_id: sensor.battery_state_of_capacity
    for:
      hours: 0
      minutes: 15
      seconds: 0
    above: 99
  - platform: numeric_state
    entity_id: sensor.inverter_input_power
    for:
      hours: 0
      minutes: 15
      seconds: 0
    above: 2000
  - platform: numeric_state
    entity_id: sensor.power_production_next_hour
    for:
      hours: 0
      minutes: 15
      seconds: 0
    above: 2500
condition:
  - condition: numeric_state
    entity_id: sensor.battery_state_of_capacity
    above: 99
  - condition: numeric_state
    entity_id: sensor.inverter_input_power
    above: 2000
  - condition: numeric_state
    entity_id: sensor.power_production_next_hour
    above: 2500
action:
  - service: water_heater.set_temperature
    data:
      temperature: 55
    target:
      entity_id: water_heater.altherma
      device_id: e054d4170e7d2be399c53b6683b05368
  - delay:
      hours: 1
      minutes: 0
      seconds: 0
      milliseconds: 0
  - service: water_heater.set_temperature
    data:
      temperature: 43
    target:
      entity_id: water_heater.altherma
      device_id: e054d4170e7d2be399c53b6683b05368
mode: queued
max: 10

```

diag attached.

[config_entry-daikin_residential_altherma-b85e79ba916040a955d14a8a6557adc8.json.txt](https://github.com/speleolontra/daikin_residential_altherma/files/10816625/config_entry-daikin_residential_altherma-b85e79ba916040a955d14a8a6557adc8.json.txt)

**Comment by jwillemsen** [comment]
Have you tried this with a different device, the error doesn't seem to relate to the water_heater. Tomorrow I have a lot of things on my schedule, will try this at some point, but probably not before Monday

**Comment by palecs** [comment]
Hello yes, with different devices it is working. No problem to w8 for Monday. Like with climate I can work with it. Leavingwateroffset is unable to set, because automation is not accepting negative values, but I can make it happen somehow. like 0 and positive, or switch to temperature sensor. But in tank it is no go for now. But no worries I will w8. Thank you for great job.

**Comment by jwillemsen** [comment]
Can you attach a script which does work with the alterma for comparison?

**Comment by palecs** [comment]
Hello but unfortunatelly not now, because it was based on previous altherma version of this integration. Issue occurs when I use in automation action service.set_ ... it fails with missing room_id, area_id or something. Or it is based on implementation of climate or water_heater in this integration not accepting additional fields from automation. 

In previous version i was able choose device.xxx_boost_xxxx and turn it on/off. By calling service it is not working. In automation action call service is mandatory to choose fields which are dropping errors. 

Thank you.

**Comment by jwillemsen** [comment]
Have you tried removing ‘device_id’

**Comment by palecs** [comment]
Hello, I was able to fix this:
- device should have name, it cannot be **unnamed device** after this when I removed device_id and area_id and used only entity_id im now able to use it in automation. so this part can be fixed

I have 2 issues now: 
- 1 in climate with automation im not able to set waterleavingofset to negative value, it is failing, so question is can be fixed from integration or it is core thing and cannot be changed?
- in water_heater implementation im not able to use boost in water tank, beause I dont see possibility to set powerfull mode on/off. Error is `Invalid tank state: powerfulMode`

Any ideas? Thank you.

**Comment by jwillemsen** [comment]
Have you tried operation_mode?

**Comment by jwillemsen** [comment]
Have you tries to add single quotes around the negative value. For both cases, enable all logging and attach the errors you get

**Comment by jwillemsen** [comment]
Have you tried this?

```
  - service: water_heater.set_operation_mode
    data:
      operation_mode: performance
```

And use '-1' (so single quotes around the negative value)?

**Comment by palecs** [comment]
Hello, so I dont know what changed but setting negative value is working without any issue and no quotes needed so maybe integration or updates fixed this.

For boost of tank please find attached debug logs and diag data.
For water_heater.set_operation_mode I have added performance as you recommended.

[home-assistant_daikin_residential_altherma_2023-02-24T08-29-50.047Z.log](https://github.com/speleolontra/daikin_residential_altherma/files/10822125/home-assistant_daikin_residential_altherma_2023-02-24T08-29-50.047Z.log)
[config_entry-daikin_residential_altherma-065f1991c9f39d9dce7a5811d97722b1.json.txt](https://github.com/speleolontra/daikin_residential_altherma/files/10822126/config_entry-daikin_residential_altherma-065f1991c9f39d9dce7a5811d97722b1.json.txt)

Thank you.

**Comment by jwillemsen** [comment]
What is the automation you use, the log complains that you send `powerfulMode` and not `performance`

**Comment by palecs** [comment]
Hello yes I understand that it complains. I put in automation performance but log says this powerfulMode. so maybe some weird mapping? 

But to logs in previous post, here is automation configuration:

```
alias: Export_teplota_bojler_boost
description: ""
trigger:
  - platform: numeric_state
    entity_id: sensor.battery_state_of_capacity
    for:
      hours: 0
      minutes: 15
      seconds: 0
    above: 99
  - platform: numeric_state
    entity_id: sensor.inverter_input_power
    for:
      hours: 0
      minutes: 15
      seconds: 0
    above: 2000
  - platform: numeric_state
    entity_id: sensor.power_production_next_hour
    for:
      hours: 0
      minutes: 15
      seconds: 0
    above: 2500
condition:
  - condition: numeric_state
    entity_id: sensor.battery_state_of_capacity
    above: 99
  - condition: numeric_state
    entity_id: sensor.inverter_input_power
    above: 2000
  - condition: numeric_state
    entity_id: sensor.power_production_next_hour
    above: 2500
action:
  - service: water_heater.set_operation_mode
    data:
      operation_mode: performance
    target:
      entity_id: water_heater.altherma
  - delay:
      hours: 0
      minutes: 30
      seconds: 0
      milliseconds: 0
  - service: water_heater.set_operation_mode
    data:
      operation_mode: heat_pump
    target:
      entity_id: water_heater.altherma
mode: queued
max: 10
```

**Comment by jwillemsen** [comment]
Please try again, fixed #98

**Comment by palecs** [comment]
Hello all seems fixed. Thank you very much superb work.
